tests/src/gui/testprocessinggui.cpp

```diff
@@ -10054,6 +10057,30 @@ void TestProcessingGui::testAlignRasterLayersWrapper()
   QCOMPARE( valueAsPythonString, QStringLiteral( "[{'inputFile': '%1','outputFile': '%2','resampleMethod': 1,'rescale': False}]" ).arg( rasterLayer->source() ).arg( layerMap["outputFile"].toString() ) );
 }
 
+void TestProcessingGui::testRasterOptionsWrapper()
+{
+  QgsProcessingParameterString param( QStringLiteral( "string" ), QStringLiteral( "string" ) );
+  param.setMetadata( {{
+      QStringLiteral( "widget_wrapper" ), QVariantMap(
+      {{QStringLiteral( "widget_type" ), QStringLiteral( "rasteroptions" ) }}
+      )
+    }
+  } );
+
+  QgsProcessingContext context;
+  QgsProcessingRasterOptionsWidgetWrapper wrapper( &param );
+
+  std::unique_ptr<QWidget> w( wrapper.createWidget() );
+  QVERIFY( w );
+
+  QSignalSpy spy( &wrapper, &QgsProcessingRasterOptionsWidgetWrapper::widgetValueHasChanged );
+  wrapper.setWidgetValue( QStringLiteral( "TFW=YES" ), context );
+  QCOMPARE( spy.count(), 1 );
+  QCOMPARE( wrapper.widgetValue().toString(), QStringLiteral( "TFW=YES" ) );
+  wrapper.setWidgetValue( QStringLiteral( "TFW=YES TILED=YES" ), context );
+  QCOMPARE( wrapper.widgetValue().toString(), QStringLiteral( "TFW=YES|TILED=YES" ) );
```

Comment:
I think we need a test here with setWidgetValue with pipe separated options too

Reply:
Added.